exercises/tom_perkins/ex01_basics/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+cmake_minimum_required ( VERSION 2.8.5 )
+
+###############################################################################
+# Setup
+###############################################################################
+# Name
+project ( WordCounter )
+
+# Find custom cmake modules
+set ( CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../CMake")
+
+# Common setup
+include ( CommonSetup )
+
+###############################################################################
+# Executable
+###############################################################################
+
+set ( SRC_FILES
+  src/main.cpp
+  # add any additional source files here
+)
+
+set ( HDR_FILES
+  # add any additional header files here
+)
+
+add_executable ( WordCounter ${SRC_FILES} ${HDR_FILES} )
```

Comment:
I got a warning on my platform because you're using c++11 features without the flags being set correctly. You can fix this by adding the following.
`set_property(TARGET WordCounter PROPERTY CXX_STANDARD 11)` **However** the RHEL6 build will continue to fail because it's using an older version of GCC, which won't support features such as the range-based for loops you used.